### PR TITLE
[FIX,DOC] Some minor fixes and revising documentation

### PR DESCRIPTION
**Pull request by rrahn**
supersedes #21

### CODE REVIEW

File: include/seqan3/io/detail/file_base.hpp [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+namespace seqan3::detail
+{
+
+// ==================================================================
+// file_base
+// ==================================================================
+
+//! The basic file class for reading and writing formatted files, e.g. "*.fasta" or "*.bam" files.
+/*!
+ *This is an abstract class and need to be implemented by a concrete file class.
```
Comment (smehringer) [inline include/seqan3/io/detail/file_base.hpp]: needs

File: include/seqan3/io/detail/file_base.hpp [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+     * This function iterates over the `valid_compression_formats` of the traits
+     * object and compares the extension identifier with the extracted compression format.
+     * If a compression format is found by a matching extension the file_name
+     * is stripped off the extension otherwise it is returned directly.
+     *
+     * \param[in] file_name The file name (path).
+     *
+     * \return file_name (stripped from valid compression format extension).
+     *
+     * \see select_compression_format
```
Comment (smehringer) [inline include/seqan3/io/detail/file_base.hpp]: why linking to the function itself? Did you mean `\see select_format` ?

File: include/seqan3/io/detail/file_base.hpp [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+     */
+    std::experimental::filesystem::path select_compression_format(std::experimental::filesystem::path & file_name);
+    //! Detects and sets the file format from the file name extension.
+    /*!
+     * This function iterates over the `valid_format_types` and checks if one of them compares
+     * equal to the file extension `ext`. 
+     * If the comparison results to `true`, then the `format` member is 
+     * set accordinlgy.
+     * If no format can be matches the function __throws__ an error.
```
Comment (smehringer) [inline include/seqan3/io/detail/file_base.hpp]: matched,

File: include/seqan3/io/detail/file_base.hpp [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+     *
+     * \param[in] ext The file name extension (e.g. ".fa", ".sam").
+     *
+     * \see select_compression_format 
+     *
+     * \throw std::runtime_error Throws if the format does not matches any of the specified `valid_format_types`.
```
Comment (smehringer) [inline include/seqan3/io/detail/file_base.hpp]: does not match

File: include/seqan3/io/sequence/sequence_file_in.hpp [hunk 5]
```diff
@@ -1,168 +1,300 @@
...
-    using valid_formats = std::variant<sequence_file_format_fasta<stream_type>,
-                                       sequence_file_format_fastq<stream_type>,
-                                       sequence_file_format_embl<stream_type>,
-                                       sequence_file_format_genbank<stream_type>,
-                                       sequence_file_format_raw<stream_type>>;
-    static constexpr std::vector<std::pair<std::string, void>> valid_compression_formats{};
+    using stream_type = std::ifstream; //!< The default stream type is std::ifstream
+    //! The default format types of the sequence file.
+    /*!
+     * The valid types are stored as an `std::variant` of formats where each format
```
Comment (smehringer) [inline include/seqan3/io/sequence/sequence_file_in.hpp]: of formats, where (comma)?

File: include/seqan3/io/sequence/sequence_file_in.hpp [hunk 6]
```diff
@@ -1,168 +1,300 @@
...
-    using sequence_file_in_traits::stream_type;               // e.g. std::ostream concept
-    using sequence_file_in_traits::valid_compression_formats; // = std::vector<std::pair<std::string, t>>
-    using sequence_file_in_traits::valid_formats;             // = std::variant<gzip_compressor, sequence_file_format_bam>
-
-    // constructor with arg
-    sequence_file_in(std::string const & _file_name);
-
-    // copy construction and assignment are deleted
-    // implicitly because we don't want multiple access to file
+    using detail::file_base<sequence_file_in_traits>::stream_type;
```
Comment (smehringer) [inline include/seqan3/io/sequence/sequence_file_in.hpp]: why no documentation? Maybe also refer to file_base::stream_type

**Comment by h-2** [comment]
@smehringer I am putting this on hold, until the other changes land, ok? Can we discuss the general sequence_file design next week?

**Comment by smehringer** [comment]
@rrahn @h-2 This looks like a stale PR that we do not need anymore..

**Comment by smehringer** [comment]
@rrahn @h-2 ping

**Pull request closed without merging**